homeassistant/components/hydrawise/entity.py

```diff
@@ -1,36 +1,33 @@
 """Base classes for Hydrawise entities."""
 
-from homeassistant.core import callback
-from homeassistant.helpers.dispatcher import async_dispatcher_connect
-from homeassistant.helpers.entity import Entity, EntityDescription
+from typing import Any
 
-from .const import SIGNAL_UPDATE_HYDRAWISE
+from homeassistant.helpers.entity import EntityDescription
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
 
 
-class HydrawiseEntity(Entity):
+class HydrawiseEntity(CoordinatorEntity):
     """Entity class for Hydrawise devices."""
 
     _attr_attribution = "Data provided by hydrawise.com"
 
-    def __init__(self, data, description: EntityDescription) -> None:
+    def __init__(
+        self,
+        *,
+        data: dict[str, Any],
+        coordinator: DataUpdateCoordinator,
+        description: EntityDescription,
+    ) -> None:
         """Initialize the Hydrawise entity."""
-        self.entity_description = description
+        super().__init__(coordinator=coordinator)
         self.data = data
+        self.api = self.coordinator.api
```

Comment:
I think you should drop the api property from the entity.
I think it adds confusion, and it is accessible already in the child entities via `self.coordinator.api`

Reply:
Done.